src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -15,11 +15,44 @@ internal class MsQuicApi : IDisposable
 
         private readonly IntPtr _registrationContext;
 
+        public static bool IsQuicSupported { get; } = GetIsQuicSupported();
+
+        private static bool GetIsQuicSupported()
+        {
+            // TODO: check using MsQuicOpen return value; but there doesn't seem to be a consistent error code for this yet.
+            OperatingSystem ver = Environment.OSVersion;
+
+            if (ver.Platform != PlatformID.Win32NT || ver.Version >= new Version(10, 0, 19041, 0))
+            {
+                try
+                {
+                    new MsQuicApi().Dispose();
+                    return true;
+                }
+                catch (QuicNotSupportedException)
+                {
+                }
+            }
+
+            return false;
+        }
+
         private unsafe MsQuicApi()
         {
-            QuicExceptionHelpers.ThrowIfFailed(
-                Interop.MsQuic.MsQuicOpen(version: 1, out MsQuicNativeMethods.NativeApi* registration),
-                "Could not open MsQuic.");
+            MsQuicNativeMethods.NativeApi* registration;
+
+            try
+            {
+                uint status = Interop.MsQuic.MsQuicOpen(version: 1, out registration);
+                if (!MsQuicStatusHelper.SuccessfulStatusCode(status))
+                {
+                    throw new QuicNotSupportedException();
```

Comment:
Yeah, if we can avoid this exception that would be nice. Just use NotSupportedException.